app/views/layouts/application.html.haml

```diff
@@ -1,6 +1,13 @@
 !!!
 %html
   %head
+    - if test?
+      :javascript
+        window.errors = []
+        window.onerror = function(a, b, c){
+          window.errors.push([a, b, c]);
+          return false; // false: don't swallow exceptions.
+        }
```

Comment:
Could you please remind me why this block of code is required? `onerror`seems an event to be triggered if an error occurs while loading an external file, but what are the `(a, b, c)` arguments?